### ISSUE TITLE
Possible to have a default value for a FormDataParam?

### ISSUE THREAD

**Issue by RohanNagar** [jersey/jersey#3652]
Hi all.

I have a method that accepts type Multipart Form Data. I want to be able to hit this endpoint sometimes without the specified `@FormDataParam("file")`, but right now when I leave it out the method returns a 400 bad request immediately. Is there a way I can set it up so that I can leave it out? Or is there some way I can set a default value for this (i.e. to null?). Any help would be appreciated. My method declaration is below:

```java
@POST
@Path("/publish")
@Consumes(MediaType.MULTIPART_FORM_DATA)
public Response publish(@Auth Key key,
                          @QueryParam("email") String email,
                          @HeaderParam("password") String password,
                          @QueryParam("type") PublishType type,
                          @QueryParam("message") String message,
                          @FormDataParam("file") InputStream inputStream,
                          @FormDataParam("file") FormDataContentDisposition
                              contentDispositionHeader,
                          @FormDataParam("title") @DefaultValue("") String videoTitle) {
// code here
}
```

Basically, I want to create an endpoint where the user can publish text to a database, and optionally include an image or some type of media. Let me know if there is another way I can achieve this.

Thank you!

**Comment by RohanNagar** [comment]
I'm closing this issue because I am currently working around it by posting empty multipart form data. This may be something that you guys could investigate in the future, perhaps allowing for `Optional` values in the form data param, and not requiring the body to exist when accepting multipart form data. For now, I will be using this workaround.
